### PR TITLE
Cleanup the readme and remove ToC

### PR DESCRIPTION
**Pull request by dnaka91**
This cleans up some details in the readme, only formatting changes and no actual difference
in the content. The cleanups include anything that was reported by `markdownlint`.

Also, I removed the table of contents as that is already auto-generated by GitHub now. This may
not be wanted if the project is the readme is mostly utilized outside of GitHub.

### CODE REVIEW

**Comment by marco-c** [comment]
Could you add markdownlint to the pre-commit configuration?

**Comment by dnaka91** [comment]
@marco-c oh I'm so sorry. I should have properly read the readme. Didn't know about that step although it's mentioned there.

I added the pre-commit step and adjusted the markdown files a bit further. Also merged the latest changes from master.

File: .markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+MD013:
+  line_length: 100
```
Comment (marco-c) [inline .markdownlint.yaml:2]: Is there a way to disable line length checks? I'm not a fan of them

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,21 +1,25 @@
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (marco-c) [inline .pre-commit-config.yaml:2]: Can you revert the whitespace-only changes in this file?

File: .markdownlint.yaml (hunk 1 above)
Comment (dnaka91) [reply .markdownlint.yaml:2]: Sure, I'll disable the lint altogether then 👍

File: .pre-commit-config.yaml (hunk 2 above)
Comment (dnaka91) [reply .pre-commit-config.yaml:2]: Will do. Sorry, vs code's auto formatting 🙇‍♂️

**Review by marco-c — APPROVED** [review]
Thanks!

**Comment by marco-c** [comment]
@dnaka91 sorry I had to revert the PR as the ToC wasn't automatically generated by GitHub. Could you submit the same PR without the ToC removal?

**Pull request merged**